stackslib/src/net/inv/nakamoto.rs

```diff
@@ -546,14 +547,18 @@ impl<NC: NeighborComms> NakamotoInvStateMachine<NC> {
             .expect("FATAL: snapshot occurred before system start");
 
         for rc in highest_rc..=tip_rc {
-            if self.reward_cycle_consensus_hashes.contains_key(&rc) {
-                continue;
+            match self.reward_cycle_consensus_hashes.entry(rc) {
+                Entry::Vacant(e) => {
+                    let Some(ch) = Self::load_consensus_hash_for_reward_cycle(sortdb, rc)? else {
+                        // NOTE: this should be unreachable, but don't panic
+                        return Err(DBError::NotFoundError.into());
+                    };
+                    e.insert(ch);
+                }
+                Entry::Occupied(_) => {
+                    continue;
```

Comment:
`continue` isn't necessary because we're at the end of the loop, how about changing this block to an `if let Entry::Vacant(e)...`